### PR TITLE
Stardew valley: backpack fix

### PR DESCRIPTION
**Pull request by agilbert1412**
## What is this fixing or adding?
Fixed a bug where playing on backpack_progression: vanilla would still generate the backpack locations in the pool
Bug was reported by @t3hf1gm3nt here: https://discord.com/channels/731205301247803413/1132878736765767781

Also fixed a bug with the item group enum. On some versions of Python 3.11 only, it was behaving strangely and duplicating the enum, leading to failed generations in most cases

## How was this tested?
1 - Reorganised the backpack unit tests for better structure and coverage, and created explicit tests to check the locations on vanilla
2 - Watch the new tests fail to prove that the bug exists
3 - Fix the bug
4 - Watch the tests now succeed
5 - Profit!

### CODE REVIEW

File: worlds/stardew_valley/test/TestBackpack.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        item_names = [item.name for item in self.multiworld.get_items()]
+        self.assertEqual(item_names.count("Progressive Backpack"), 2)
+
+    def test_2_backpack_locations(self):
+        location_names = {location.name for location in self.multiworld.get_locations()}
+        self.assertIn("Large Pack", location_names)
+        self.assertIn("Deluxe Pack", location_names)
+
+    def test_progressive_backpack_is_in_early_pool(self):
+        self.assertIn("Progressive Backpack", self.multiworld.early_items[1])
```
Comment (black-sliver) [inline worlds/stardew_valley/test/TestBackpack.py:44]: Please add newline at end of file here and ...

File: worlds/stardew_valley/test/mods/TestBiggerBackpack.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        self.assertEqual(item_names.count("Progressive Backpack"), 3)
+
+    def test_3_backpack_locations(self):
+        location_names = {location.name for location in self.multiworld.get_locations()}
+        self.assertIn("Large Pack", location_names)
+        self.assertIn("Deluxe Pack", location_names)
+        self.assertIn("Premium Pack", location_names)
+
+    def test_progressive_backpack_is_in_early_pool(self):
+        self.assertIn("Progressive Backpack", self.multiworld.early_items[1])
```
Comment (black-sliver) [inline worlds/stardew_valley/test/mods/TestBiggerBackpack.py:51]: ... here

**Comment by ThePhar** [comment]
adding release/blocker as this fixes a sample issue too apparently on RC1: https://discord.com/channels/731205301247803413/731214280439103580/1133164985506340874

**Review by black-sliver — APPROVED** [review]
Fixes two bugs, so should definitely be merged for RC2.

lgtm other than too many tests :D

**Pull request merged**